stock_picking_operation_quick_change/wizards/stock_picking_wizard.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# © 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPickingOperationWizard(models.TransientModel):
+    _name = 'stock.picking.operation.wizard'
+
+    def _prepare_default_values(self, picking):
+        return {'location_dest_id': picking.location_dest_id.id}
+
+    @api.model
+    def default_get(self, fields):
+        res = super(StockPickingOperationWizard, self).default_get(fields)
+        active_model = self.env.context['active_model']
+        active_ids = self.env.context['active_ids'] or []
+        picking = self.env[active_model].browse(active_ids)
+        res.update(self._prepare_default_values(picking))
+        return res
+
+    def _default_old_dest_location_id(self):
+        stock_picking_obj = self.env['stock.picking']
+        pickings = stock_picking_obj.browse(self.env.context['active_ids'])
+        first_operation = pickings.mapped('pack_operation_product_ids')[:1]
+        return first_operation.location_dest_id.id
+
+    def _get_allowed_locations(self):
+        return ['internal']
+
+    def _get_allowed_picking_states(self):
+        return ['assigned']
+
+    location_dest_id = fields.Many2one(
+        comodel_name='stock.location',
+        string='Actual destination location',
+        required=True,
+        readonly=True,
+    )
+    old_location_dest_id = fields.Many2one(
+        comodel_name='stock.location',
+        string='Old destination location',
+        default=_default_old_dest_location_id,
+        domain=lambda self: [('usage', 'in', self._get_allowed_locations())],
```

Comment:
Maybe the domain could be entirely get through method ('_get_allowed_location_domain')? Same for new_location_dest_id